### ISSUE TITLE
Building of release files should be automated

### ISSUE THREAD

**Issue by jermar** [HelenOS/helenos#108]
**Reported by svoboda on 3 Aug 2009 21:46 UTC**
There should be a script to build the release files. This should be easy to do and has the following advantages:

 * Faster, easier building is good especially when releasing in quick succession (e.g. bugfix release).
 * More deterministic, less prone to errors.
 * More transparent (anyone can see how the builds are made, so they know exactly what they are getting)
 * Susceptible to testing.

Basically it should be possible to configure HelenOS from one of the preset files via command line (this is very useful itself). Then, the script would just select the proper configuration for each release file and then copy it it from the build directory.

**Comment by jermar** [comment]
**Modified by jermar on 16 Dec 2009 11:59 UTC**

**Comment by jermar** [comment]
**Modified by jermar on 5 Dec 2010 18:50 UTC**

**Comment by jermar** [comment]
**Modified by jermar on 27 Jan 2011 19:43 UTC**

**Comment by jermar** [comment]
**Modified by jermar on 28 Jan 2011 13:17 UTC**

**Comment by jermar** [comment]
**Modified by jermar on 28 Jan 2011 14:06 UTC**

**Comment by jermar** [comment]
**Modified by jermar on 2 Mar 2011 19:17 UTC**

**Comment by jermar** [comment]
**Comment by svoboda on 7 Mar 2011 19:56 UTC**
With the integration of changeset:mainline,863 and changeset:mainline,879 most of this problem is solved. Still we need to figure out what additional tuning should be applied by the release Makefile so that the release files are fit for release.

* for example, the Bazaar revision should not be compiled in

**Comment by jermar** [comment]
**Comment by svoboda on 17 Mar 2011 07:53 UTC**
In changeset:mainline,881, changeset:mainline,882 and changeset:mainline,883 I made the release Makefile first export the source from Bazaar and then build a source tarball and build the release images. This also ensures the Bazaar revision is not built into the release images. I consider this ticket resolved.

In the future it would be nice to unify 'make check' and 'make release' to avoid duplication.
